python/plugins/processing/tests/GdalAlgorithmsMemoryTest.py

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*-
+
+"""
+***************************************************************************
+    GdalAlgorithmsMemoryTest
+    ---------------------
+    Date                 : October 2016
+    Copyright            : (C) 2016 by Germán Carrillo
+    Email                : gcarrillo  at linuxmail dot org
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+__author__ = 'Germán Carrillo'
+__date__ = 'October 2016'
+__copyright__ = '(C) 2016, Germán Carrillo'
+
+# This will get replaced with a git SHA1 when you do a git archive
+
+__revision__ = '$Format:%H$'
+
+from qgis.testing import start_app, unittest
+from processing.core.Processing import Processing
+from processing.tests.TestData import points2, polygons, polygonsGeoJson
+from processing.tools import vector, dataobjects, general
+from qgis.core import (QgsVectorLayer, QgsFeatureRequest, QgsGeometry,
+                       QgsDataSourceURI, QgsField)
+from processing.core.ProcessingConfig import ProcessingConfig
+from PyQt4.QtCore import QVariant
+
+import os.path
+import errno
+import shutil
+
+dataFolder = os.path.join(os.path.dirname(__file__), 'testdata/')
+expectedFolder = os.path.abspath(os.path.join(dataFolder, 'expected', 'gdal'))
+tmpBaseFolder = os.path.join(os.sep, 'tmp', 'qgis_test', str(os.getpid()))
```

Comment:
Better use a portable approach like the [one here](https://github.com/gacarrillor/QGIS/blob/a375f6fcd7665cdbdd50e4f014c40cc1e66f3bb9/python/plugins/processing/tests/AlgorithmsTestBase.py#L154-L158)